watertap/unit_models/zero_order/gac_zo.py

```diff
@@ -37,7 +37,74 @@ def build(self):
         self._tech_type = "gac"
 
         build_sido(self)
-        self._Q = Reference(self.properties_in[:].flow_vol)
-        # TODO: incorporate a*Q**b relationship for electricity consumption based on EPA data fitting;
-        # apply pump electricity consumption in the meantime
-        pump_electricity(self, self._Q)
+
+        # Empty Bed Contact Time
+        self.empty_bed_contact_time = Var(
+            units=pyunits.hour,
+            bounds=(0, None),
+            doc="Empty bed contact time of unit")
+
+        self._fixed_perf_vars.append(self.empty_bed_contact_time)
+        self._perf_var_dict["Empty Bed Contact Time"] = \
+            self.empty_bed_contact_time
+
+        # Electricity Demand
+        self.electricity = Var(self.flowsheet().time,
+                               units=pyunits.kW,
+                               bounds=(0, None),
+                               doc="Electricity consumption of unit")
+
+        self.volumetric_electricity_intensity = Var(
+            units=pyunits.kW/pyunits.m**3,
+            doc="Electricity intensity with respect to inlet flowrate of unit")
+
+        self.energy_electric_flow_vol_inlet = Var(
+            units=pyunits.kWh/pyunits.m**3,
+            doc="Electricity intensity with respect to inlet flowrate of unit")
+
+        @self.Constraint(
+            doc='Electricity intensity based on empty bed contact time.')
+        def electricity_intensity_constraint(b):
+            return (b.energy_electric_flow_vol_inlet ==
+                    b.volumetric_electricity_intensity *
+                    b.empty_bed_contact_time)
```

Comment:
Where does getting energy from electricity_intensity*EBCT come from? Is that in WT3? I don't think I have seen this relationship before.

Reply:
That is from the EPA cost curves - the data shows that intensity is a function of `ebct`, however with only two data points the best we can do is a linear relationship.

Reply:
There should be three data points for each curve. But the EBCT column in that dataset is only supposed to be used for selecting data. We fit the electricity intensity data to E = A * Q ** B.